spec/lib/sarif/language_version/language_version_scanner_sarif_spec.rb

```diff
@@ -0,0 +1,100 @@
+                                                               error_msg|
+  describe '#build_runs_object' do
+    let(:config_file) { YAML.load_file(invalid_version_repo_path + '/salus.yml') }
+    let(:scanner) do
+      scanner_class.new(repository: repo,
+                    config: config_file['scanner_configs'][scanner_name.to_s])
+    end
+    before { scanner.run }
+
+    context 'version mismatch found in project' do
+      let(:path) { invalid_version_repo_path }
+      let(:repo) { Salus::Repo.new(path) }
+
+      it 'creates valid sarif report with results populated' do
+        adapter = described_class.new(scanner.report, path)
+        report = adapter.build_runs_object(true)
+        rules = report['tool'][:driver]['rules']
+        results = report['results']
+
+        expect(results.size).to eq(1)
+        expect(results).to include(
+          {
+            "ruleId": "LV0001",
+            "ruleIndex": 0,
+            "level": "error",
+            "message": {
+              "text": error_msg
+            },
+            "properties": { "severity": "HIGH" },
+            "locations": [
+              {
+                "physicalLocation": {
+                  "artifactLocation": {
+                    "uri": "",
+                    "uriBaseId": "%SRCROOT%"
+                  }
+                }
+              }
+            ]
+          }
+        )
+
+        doc = "https://github.com/coinbase/salus/blob/master/docs/scanners/language_version_scan.md"
+        expect(rules[0]).to include(
+          {
+            "id": "LV0001",
+            "name": scanner_name,
+            "fullDescription": {
+              "text": error_msg
+            },
+            "messageStrings": {},
+            "helpUri": doc,
+            "help": {
+              "text": "More info: #{doc}",
+              "markdown": "[More info](#{doc})."
+            }
+          }
+        )
+      end
+
+      it 'creates a sarif report that follows the schema' do
+        report = Salus::Report.new(project_name: "Neon Genesis")
+        scan_reports = [
+          [scanner.report, false]
+        ]
+        sarif_json = Sarif::SarifReport.new(scan_reports).to_sarif
+        filtered_sarif = report.apply_report_sarif_filters(sarif_json)
+        expect { Sarif::SarifReport.validate_sarif(filtered_sarif) }.not_to raise_error
+      end
+    end
+
+    context 'no vulnerabilites found in project' do
+      let(:config_path) { valid_version_repo_path + '/salus.yml' }
+      let(:config_file) { YAML.load_file(config_path) }
+      let(:path) { valid_version_repo_path }
+      let(:repo) { Salus::Repo.new(path) }
+
+      it 'does not find any version mismatch for valid repository' do
+        adapter = described_class.new(scanner.report, path)
+        report = adapter.build_runs_object(true)
+        rules = report['tool'][:driver]['rules']
+        results = report['results']
+
+        expect(results.size).to eq(0)
```

Comment:
Minor: Optional version reads a little better (your discretion on if you wanna change) - it's MINOR
 
expect(results).to be_empty
expect(rules).to be_empty

Reply:
Updated.